### PR TITLE
Refactor task module using pyscopg and pydantic

### PR DESCRIPTION
**Pull request by Pradip-p**
Continuing this PR: https://github.com/hotosm/drone-tm/pull/128

Updated Task Modules using psycopg by replacing encode/databases

### CODE REVIEW

File: src/backend/app/tasks/task_routes.py [hunk 1]
```diff
@@ -53,9 +54,8 @@ async def new_event(
 
     match detail.event:
         case EventType.REQUESTS:
-            # TODO: Combine the logic of `update_or_create_task_state` and `request_mapping` functions into a single function if possible. Will do later.
             project = await get_project_by_id(db, project_id)
```
Comment (spwoodcock) [inline src/backend/app/tasks/task_routes.py]: Could we possibly get the project from `project.deps.get_project_by_id`?
Using `Depends`?

The project_crud.get_project_by_id function can probably be deleted

**Review by spwoodcock — CHANGES_REQUESTED** [review]
Small change, but otherwise looks excellent!

**Comment by spwoodcock** [comment]
Just saw your latest changes!

The distinction between logic that belongs on the schemas and logic in the `crud` file is blurry.

I think we should:
- Put these functions directly on the model in schemas for sure: `one`, `all`, `create`, `update`, `delete` (possibly others we think of, like `create_or_update`, but that will do for now).
- Then the `xxx_crud.py` files can be renamed `xxx_logic.py` and we put any endpoint logic here, such as `request_mapping` for example.

File: src/backend/app/tasks/task_routes.py (hunk 1 above)
Comment (Pradip-p) [reply src/backend/app/tasks/task_routes.py]: Yes, we can indeed get the project using project.deps.get_project_by_id with Depends. I’ve now removed the project_crud.get_project_by_id function as it is redundant. Thanks for pointing that out!

**Review by nrjadkry — APPROVED** [review]

**Pull request merged**